pinot-plugins/pinot-input-format/pinot-avro-base/src/main/java/org/apache/pinot/plugin/inputformat/avro/AvroUtils.java

```diff
@@ -368,6 +341,21 @@ public static Object handleMultiValue(@Nullable Collection values) {
     for (Object value : values) {
       list.add(handleSingleValue(value));
     }
-    return list;
+    return RecordReaderUtils.convertMultiValue(list);
+  }
+
+  /**
+   * Converts the values withing the map to single-valued values
+   */
+  public static Object handleMap(@Nullable Map map) {
```

Comment:
is this specific to Avro?

Reply:
Right now, yes. It is designed based off what was done in AvroUtils to handle the map